App/Controls/SizedFrame.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using Windows.Foundation;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+
+namespace Coder.Desktop.App.Controls;
+
+public class SizedFrameEventArgs : EventArgs
+{
+    public Size NewSize { get; init; }
+}
+
+/// <summary>
+///     SizedFrame extends Frame by adding a SizeChanged event. Sadly this is necessary because
```

Comment:
SizedFrame extends Frame by adding a SizeChanged event, and wiring up the event to trigger whenever:

 * The contained Page changes size
 * We switch to a different page

Sadly, this is necessary...